R/pagination.R

```diff
@@ -1161,16 +1161,21 @@ paginate_to_mpfs <- function(obj,
   ## but we will if we ever allow force_paginate to do horiz
   ## pagination.
   if (length(fpags) > 1) {
-    deep_pag <- lapply(
-      fpags, paginate_to_mpfs,
+    # Correction for case we are entering here (page_by?)
+    if (!is.null(page_num)) {
+      prov_footer(obj) <- head(prov_footer(obj), -1)
+      fpags <- do_forced_paginate(obj)
+    }
+    deep_pag <- paginate_to_mpfs( # what about the other parameters?
+      fpags,
       pg_size_spec = pg_size_spec,
       colwidths = colwidths,
       min_siblings = min_siblings,
       nosplitin = nosplitin,
       verbose = verbose,
       page_num = page_num
     )
-    return(unlist(deep_pag, recursive = FALSE))
```

Comment:
this is a breaking change, previously `paginate_to_mpfs` was guaranteed to return a shallow list of pages, now it can return a nested list